### PR TITLE
change vetoer update to be 2 step process

### PR DESCRIPTION
**Pull request by davidbrai**
this is in order to reduce the chances of setting the vetoer
to the wrong address

### CODE REVIEW

**Comment by davidbrai** [comment]
currently getting warning:
"Warning: Contract code size is 24924 bytes and exceeds 24576 bytes"

File: packages/nouns-contracts/test/foundry/NounsDAOLogicV2.t.sol [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                        minQuorumVotesBPS: 200,
+                        maxQuorumVotesBPS: 2000,
+                        quorumCoefficient: 10000
+                    })
+                )
+            )
+        );
+    }
+
+    function test_updateVetoer_failsIfNotCurrentVetoer() public {
```
Comment (eladmallel) [inline packages/nouns-contracts/test/foundry/NounsDAOLogicV2.t.sol:44]: maybe rename the test functions to `test_setPendingVetoer_scenario`?

File: packages/nouns-contracts/test/foundry/NounsDAOLogicV2.t.sol [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        vm.expectRevert(NounsDAOLogicV2.VetoerOnly.selector);
+        daoProxy._setPendingVetoer(address(0x1234));
+    }
+
+    event NewPendingVetoer(address oldPendingVetoer, address newPendingVetoer);
```
Comment (eladmallel) [inline packages/nouns-contracts/test/foundry/NounsDAOLogicV2.t.sol:49]: nit: move events to the top of the contract

File: packages/nouns-contracts/test/foundry/NounsDAOLogicV2.t.sol (hunk 1 above)
Comment (davidbrai) [reply packages/nouns-contracts/test/foundry/NounsDAOLogicV2.t.sol]: done

File: packages/nouns-contracts/test/foundry/NounsDAOLogicV2.t.sol (hunk 2 above)
Comment (davidbrai) [reply packages/nouns-contracts/test/foundry/NounsDAOLogicV2.t.sol]: done

File: packages/nouns-contracts/test/foundry/NounsDAOLogicV2.t.sol [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+        vm.expectRevert(NounsDAOLogicV2.PendingVetoerOnly.selector);
+        daoProxy._acceptVetoer();
+
+        vm.prank(pendingVetoer);
+        vm.expectEmit(true, true, true, true);
+        emit NewVetoer(vetoer, pendingVetoer);
+        daoProxy._acceptVetoer();
+
+        assertEq(daoProxy.vetoer(), pendingVetoer);
```
[the remark this answers is not in the document]
Comment (davidbrai) [reply packages/nouns-contracts/test/foundry/NounsDAOLogicV2.t.sol:79]: done

File: packages/nouns-contracts/contracts/governance/NounsDAOLogicV2.sol [hunk 4]
```diff
@@ -833,26 +862,43 @@ contract NounsDAOLogicV2 is NounsDAOStorageV2, NounsDAOEventsV2 {
...
+        }
+
+        emit NewPendingVetoer(pendingVetoer, newPendingVetoer);
+
+        pendingVetoer = newPendingVetoer;
+    }
 
-        emit NewVetoer(vetoer, newVetoer);
+    function _acceptVetoer() external {
+        if (msg.sender != pendingVetoer || pendingVetoer == address(0)) {
```
Comment (solimander) [inline packages/nouns-contracts/contracts/governance/NounsDAOLogicV2.sol:879]: `pendingVetoer == address(0)` will never be evaluated, right? If `pendingVetoer` is `address(0)`, `msg.sender` will never be `pendingVetoer`.

**Comment by solimander** [comment]
Currently, the vetoer could 'burn' the veto power without actually burning the veto by setting a pending vetoer, burning the veto, and calling the accept vetoer function. Probably a good idea to close the loophole.

File: packages/nouns-contracts/contracts/governance/NounsDAOLogicV2.sol (hunk 4 above)
Comment (davidbrai) [reply packages/nouns-contracts/contracts/governance/NounsDAOLogicV2.sol:879]: @solimander yes I agree. I was mostly following the same pattern as in [_acceptAdmin](https://github.com/nounsDAO/nouns-monorepo/blob/verbs-nip-1-save-proposal-creation-block/packages/nouns-contracts/contracts/governance/NounsDAOLogicV2.sol#L819)
but I agree it's probably redundant.
do you see any reason for the `msg.sender != address(0)` in the `_acceptAdmin` function?

**Comment by davidbrai** [comment]
@solimander good catch on the pending vetoer not being reset.
fixed here: https://github.com/nounsDAO/nouns-monorepo/pull/554/commits/ce985240d229575cd0eb87ef62057d27afdf8665

File: packages/nouns-contracts/contracts/governance/NounsDAOLogicV2.sol (hunk 4 above)
Comment (solimander) [reply packages/nouns-contracts/contracts/governance/NounsDAOLogicV2.sol:879]: Nope, it's redundant in both cases. Only protects against the unfeasible scenario in which the private key to the zero address is found.

Comment (davidbrai) [reply packages/nouns-contracts/contracts/governance/NounsDAOLogicV2.sol]: OK, removed it from _acceptVetoer: https://github.com/nounsDAO/nouns-monorepo/pull/554/commits/3a36e0ae6ffebe3c415f3f20c70988c97575901f

**Review by solimander — APPROVED** [review]

**Pull request merged**